### PR TITLE
encapsulate operation about meta file

### PR DESCRIPTION
**Pull request by july2993**
ref #557 

encapsulate operation about meta file in pkg/meta/spec*

### CODE REVIEW

File: pkg/dm/spec/cluster.go [hunk 1]
```diff
@@ -34,42 +30,7 @@ type DMMeta struct {
...
-		return nil, errors.Trace(err)
-	}
-
-	if err = yaml.Unmarshal(yamlFile, &cm); err != nil {
-		return nil, errors.Trace(err)
-	}
-	return &cm, nil
+// NewDMSpec create dm Spec.
+func NewDMSpec() *meta.Spec {
+	return meta.NewSpec(filepath.Join(cspec.ProfileDir(), "dm"))
```
Comment (lonng) [inline pkg/dm/spec/cluster.go:35]: Prefer to define a constant for `dm` string literal.

File: pkg/meta/spec.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
+package meta
```
Comment (lonng) [inline pkg/meta/spec.go:1]: The license header is missing.

File: pkg/meta/spec_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
+package meta
```
Comment (lonng) [inline pkg/meta/spec_test.go:1]: ditto

File: pkg/meta/spec.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+	// MetaFileName is the file name of the meta file.
+	MetaFileName = "meta.yaml"
+	// PatchDirName is the directory to store patch file eg. {PatchDirName}/tidb-hotfix.tar.gz
+	PatchDirName = "patch"
+	// BackupDirName is the directory to save backup files.
+	BackupDirName = "backup"
+)
+
+// Spec control management of spec meta data.
+type Spec struct {
```
Comment (lonng) [inline pkg/meta/spec.go:35]: Prefer to choose another name, e.g: `SpecManager`.

File: pkg/meta/spec.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+func NewSpec(base string) *Spec {
+	return &Spec{
+		base: base,
+	}
+}
+
+// ClusterPath returns the full path to a subpath (file or directory) of a
+// cluster, it is a subdir in the profile dir of the user, with the cluster name
+// as its name.
+func (s *Spec) ClusterPath(cluster string, subpath ...string) string {
```
Comment (lonng) [inline pkg/meta/spec.go:49]: It's a good idea to abstract a new type to organize metadata, I think all exported methods name should refactor and shouldn't contain the `Cluster` .

File: components/cluster/command/root.go [hunk 6]
```diff
@@ -49,6 +50,8 @@ var (
 	skipConfirm bool
 )
 
+var tidbSpec *meta.SpecManager
```
Comment (lonng) [inline components/cluster/command/root.go:53]: I think the `tidbSpec` should the global variable of `pkg/cluster/spec` and make it reusable in https://github.com/pingcap/tiup/pull/567/files#diff-fdca3c2d2bcb7afb276475bee9c63c47R51

File: components/dm/command/root.go [hunk 7]
```diff
@@ -42,6 +44,8 @@ var (
 	skipConfirm bool
 )
 
+var dmspec *meta.SpecManager
```
Comment (lonng) [inline components/dm/command/root.go:47]: ditto

File: components/cluster/command/root.go (hunk 6 above)
Comment (july2993) [reply components/cluster/command/root.go]: resolve in https://github.com/pingcap/tiup/pull/567/commits/a82ee6c3f48fd751629f13db687915dc646e3c91

**Pull request merged**